BaseClasses.py

```diff
@@ -776,9 +776,11 @@ def _update_reachable_regions_explicit_indirect_conditions(self, player: int, qu
                 self.path[new_region] = (new_region.name, self.path.get(connection, None))
 
                 # Retry connections if the new region can unblock them
-                for new_entrance in self.multiworld.indirect_connections.get(new_region, set()):
-                    if new_entrance in blocked_connections and new_entrance not in queue:
-                        queue.append(new_entrance)
+                entrances = self.multiworld.indirect_connections.get(new_region)
+                if entrances is not None:
+                    relevant_entrances = entrances.intersection(blocked_connections)
+                    relevant_entrances.difference_update(queue)
+                    queue.extend(relevant_entrances)
```

Comment:
```suggestion
                try:
                    entrances = self.multiworld.indirect_connections[new_region]
                    relevant_entrances = entrances.intersection(blocked_connections)
                    relevant_entrances.difference_update(queue)
                    queue.extend(relevant_entrances)
                except KeyError:
                    pass
```
is this faster? my understanding is dict.get() essentially does this same thing so this just skips the `is not None` check

Reply:
No, catching exceptions is slow. Exceptions are not meant to be a default codepath, they are for handling exceptional circumstances.

If the expectation was that almost all regions would be present in `self.multiworld.indirect_connections`, and that a region not being present was an 'exceptional circumstance', then the try-except could be faster. It really just depends on what percentage of the time you are expecting to have to catch an exception.

My understanding of `dict.get()` is that it is pretty close to `my_dict[key] if key in my_dict`, but slightly faster for types with more complex, non-cached hashing because `dict.get()` should only need to get the hash of `key` once when the key is present in the dictionary.